doc/compressed_transactions.md

```diff
@@ -0,0 +1,153 @@
+| Hash Type               | 1 Byte           | If the Hash Type for this Input was non standard, contains a single byte. |
+| Txid Blockheight        | Bitstream VarInt | If 0 indicates a coinbase input, Otherwise contains the offset from Minimum Blockheight for the Txid. |
+| Txid/Signature Data     | Variable         | Txid/Signatures are determined to be uncompressed either by the output script of the previous transaction, or if the Txid Blockheight is zero. For each Compressed Txid/Signature See Section 3.6.1. For each Uncompressed Txid/Signature See Section 3.6.2. |
+
+### 3.6.1 Compressed Txid/Signature Data
+
+| Name              | Width            | Description |
+| ------------------|------------------|-------------|
+| Txid Block Index  | Bitstream VarInt | Contains the flattend index from the Txid Blockheight for the Vout. |
+| Signature         | 64 Bytes         | Contains the 64 byte signature. |
+
+### 3.6.2 Uncompressed Txid/Signature Data
+
+| Name                    | Width         | Description |
+| ------------------------|---------------|-------------|
+| Txid                    | 32 Bytes      | Contains the 32 byte Txid. |
+| Signature               | VLP-Bytestream | Contains the signature. |
+
+### 3.7 Output Data
+
+| Name          | Width            | Description |
+| --------------|------------------|-------------|
+| Script        | VLP-Bytestream   | Contains the output script. |
+| Amount        | Bitstream VarInt | A Bitstream VarInt containing the Output Amount. |
+
+## 4 Ideal Transaction
+
+The target transaction for the most optimal compression was chosen
+based off the most common transactions that are likely to be used
+for purposes that requires the best compression.
+
+| Field              | Value                             | Savings                           |
+|--------------------|-----------------------------------|-----------------------------------|
+| Version            | Less than four                    | 30 Bits                           |
+| Input Count        | Less then four                    | 30 Bits                           |
+| Output Count       | Less then four                    | 30 Bits                           |
+| LockTime           | 0                                 | 30 Bits                           |
+| Standard Hash Type | 0x00 for Taproot, 0x01 for Legacy | 7 Bits For Each Input             |
+| Standard Sequence  | 0x00, 0xFFFFFFFE, or 0xFFFFFFFF   | 62 Bits For Each Input            |
+| Txid/Vout          | Compressed Txid                   | 32 Bytes For Each Input           |
+| Input Signature    | Non-custom Script Signing         | 40-72 Bytes For Each Legacy Input |
+| Output Scripts     | Non-custom Scripts                | 2-5 Bytes For Each Output         |
+| Output Amount      | No Restrictions                   | 0-7 Bytes For Each Output         |
+| Total              | 2-input 2-output Transaction      | 100-124 Bytes and 2 Bits          |
```

Comment:
In ddde5c7defae80058c35bcd5fdc27ee57935495f:

This is really impressive but I don't think it's right. Each input needs (at least) a 64-byte signature, right? So the minimum size is 128?

Reply:
That is the total savings of a transaction if it has 2 inputs and 2 outputs it can be compressed by at least 100 and up to 124 bytes for non-legacy transactions, Ive added a field for legacy transactions which are at least 140 and up to 196 Bytes of compression.

Reply:
Ah, total *savings*. Not total transaction size.

Maybe we want to remove the "total" line and just provide an example transaction with before and after sizes.

Reply:
Fixed, Thanks!